packages/sdk-client/src/api/api-client-options-helper.ts

```diff
@@ -1,37 +1,87 @@
-// import { ApiTokenRequest, Oauth2TokenRequest, SigningRequest, XTimestampRequest } from '../plugins';
-// // import { ApiClientOptions } from './api-client-options';
-//
-// export const buildApiClientOptionsForProjectId = (
-//   projectId: string,
-//   keyId: string,
-//   keySecret: string,
-// ) => {
-//   return {
-//     projectId,
-//     requestPlugins: [new Oauth2TokenRequest(keyId, keySecret)],
-//     useServicePlanId: false,
-//   };
-// };
-//
-// export const  buildApiClientOptionForServicePlanId = (
-//   servicePlanId: string,
-//   apiToken: string,
-// ) => {
-//   return {
-//     projectId: servicePlanId,
-//     requestPlugins: [new ApiTokenRequest(apiToken)],
-//     useServicePlanId: true,
-//   };
-// };
-//
-// export const buildApiClientOptionForApplication = (
-//   applicationKey: string,
-//   applicationSecret: string,
-// ) => {
-//   return {
-//     requestPlugins: [
-//       new XTimestampRequest(),
-//       new SigningRequest(applicationKey, applicationSecret),
-//     ],
-//   };
-// };
+import { Region, SinchClientParameters } from '../domain';
+import { ApiClientOptions } from './api-client-options';
+import {
+  ApiTokenRequest,
+  Oauth2TokenRequest,
+  SigningRequest,
+  XTimestampRequest,
+} from '../plugins';
+
+export const buildOAuth2ApiClientOptions = (params: SinchClientParameters, apiName: string): ApiClientOptions => {
+  if (!params.projectId || !params.keyId || !params.keySecret) {
+    throw new Error(`Invalid configuration for the ${apiName} API: "projectId", "keyId" and "keySecret" values must be provided`);
+  }
+  const apiClientOptions: ApiClientOptions = {
+    projectId: params.projectId,
+    requestPlugins: [new Oauth2TokenRequest(params.keyId, params.keySecret, params.authHostname)],
+    useServicePlanId: false,
+  };
+  addPlugins(apiClientOptions, params);
+  return apiClientOptions;
+};
+
+export const buildApplicationSignedApiClientOptions = (
+  params: SinchClientParameters, apiName: string,
+): ApiClientOptions => {
+  if (!params.applicationKey || !params.applicationSecret) {
+    throw new Error(`Invalid configuration for the ${apiName} API: "applicationKey" and "applicationSecret" values must be provided`);
+  }
+  const apiClientOptions: ApiClientOptions = {
+    requestPlugins: [
+      new XTimestampRequest(),
+      new SigningRequest(params.applicationKey, params.applicationSecret),
+    ],
+  };
+  addPlugins(apiClientOptions, params);
+  return apiClientOptions;
+};
+
+export const buildFlexibleOAuth2OrApiTokenApiClientOptions = (
+  params: SinchClientParameters, region: Region, apiName: string,
+): ApiClientOptions => {
+  let apiClientOptions: ApiClientOptions | undefined;
+  // Check the region: if US or EU, try to use the OAuth2 authentication with the access key / secret under the project Id
+  if (!params.forceServicePlanIdUsageForSmsApi
+    && (region === Region.UNITED_STATES || region === Region.EUROPE)) {
```

Comment:
typescript question for my own growth:
When calling `buildFlexibleOAuth2OrApiTokenApiClientOptions` with `region` parameter value as string `us` (and **not** `Region.UNITED_STATES`)
Is the test result will be as expected here ?

Reply:
It would be the same result as by assigning string values to enum members, Typescript infers these values as string literals, so in any case, we are comparing strings with strings